src/vos/vos_io.c

```diff
@@ -2347,12 +2347,8 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 	}
 
 	if (err == 0 && ioc->ic_epr.epr_hi > ioc->ic_obj->obj_df->vo_max_write) {
-		if (DAOS_ON_VALGRIND)
-			err = umem_tx_xadd_ptr(umem, &ioc->ic_obj->obj_df->vo_max_write,
-					       sizeof(ioc->ic_obj->obj_df->vo_max_write),
-					       UMEM_XADD_NO_SNAPSHOT);
-		if (err == 0)
-			ioc->ic_obj->obj_df->vo_max_write = ioc->ic_epr.epr_hi;
+		umem_atomic_copy(umem, &ioc->ic_obj->obj_df->vo_max_write, &ioc->ic_epr.epr_hi,
+				 sizeof(ioc->ic_obj->obj_df->vo_max_write), UMEM_COMMIT_DEFER);
```

Comment:
For BMEM this is equivalent to a tx_add_range(), but for PMEM, it's a immediate flush, which looks worse than a tx_add_range(), why don't we just call a tx_add_range() instead?

Reply:
It's not equivalent to tx_add_range for BMEM because there is no undo log.

For pmem, we could do what I ad before with tx_add_range with no snapshot.

Reply:
Should we do tx_add_range() without snapshot for both BMEM and PMEM?

Reply:
we could but in pmem, this is a noop aside from shutting up valgrind, if it's running.   For bmem, we would need it to behave similarly to current umem_atomic_copy with the DEFER flag